airbyte-integrations/connectors/source-zendesk-talk/source_zendesk_talk/api.py

```diff
@@ -0,0 +1,288 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import sys
+from abc import ABC, abstractmethod
+from functools import partial
+from http import HTTPStatus
+from typing import Any, Callable, Iterable, Iterator, List, MutableMapping, Union
+
+import backoff
+import pendulum as pendulum
+import requests
+from base_python.entrypoint import logger
+
+from .errors import ZendeskAccessDenied, ZendeskInvalidAuth, ZendeskRateLimited, ZendeskTimeout
+
+
+def retry_pattern(backoff_type, **wait_gen_kwargs):
+    def sleep_on_ratelimit(details):
+        _, exc, _ = sys.exc_info()
+        logger.info(str(exc))
+        logger.info(f"Caught retryable error after {details['tries']} tries. Waiting {details['wait']} more seconds then retrying...")
+
+    def log_giveup(_details):
+        logger.error("Max retry limit reached")
+
+    return backoff.on_exception(
+        backoff_type,
+        ZendeskRateLimited,
```

Comment:
how come this doesn't include `ZendeskTimeout`?

Reply:
Agree with you, added an error to the backoff tuple